### DISCUSSION TITLE
(Inbound) Webhooks: Add option for custom trigger path.

### DISCUSSION THREAD

**Discussion by scolastico** [directus/directus#15722] [Feature Requests]
I think it would be very nice to have the possibility to change the path of the webhooks to something custom.
Because I think it would be better to call for example `https://example.com/flows/trigger/special_action` instead of `https://example.com/flows/trigger/2fff1478-d024-404e-bd1f-b5d8c738d581`. Either for programming or maintenance I see advantages there.

**Comment by scolastico** [comment]
Also to go into more detail, I would implement it so that these custom paths cannot be in a uuid format (to avoid for duplicate mapping) and that they must be unique.

**Comment by azrikahar** [comment]
It was definitely something we'd like to have in the early stages of Flows, mentioned over here: https://github.com/directus/directus/discussions/12053#discussioncomment-2500987 and was a nice-to-have item noted here: https://github.com/directus/directus/pull/12522#pullrequestreview-944880310 (quote below)

>  A route alias would be great (so I can call it `/flows/trigger/sync-with-stripe` etc)

but it just didn't make the cut (of an already large feature! 😄). That said, I'll have to defer to @nickrum on the feasibility and lift required for such implementation.

**Reply by nickrum** [reply]
Shouldn't be too hard to implement. We simply went with the UUID approach because it was the easiest, as that way we didn't have to check for uniqueness. I'm wondering if it would be enough to be able to trigger a flow by name instead of by UUID.

If I remember correctly, there was also a bigger discussion regarding some sort of "URL builder" that would allow you to fully customize the URLs DIrectus is using 🤔

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
